contracts/LightClientBridge.sol

```diff
@@ -0,0 +1,282 @@
+     */
+    function completeSignatureCommitment(
+        uint256 id,
+        bytes32 payload,
+        bytes[] memory signatures,
+        uint256[] memory validatorPositions,
+        address[] memory validatorPublicKeys,
+        bytes32[][] memory validatorPublicKeyMerkleProofs
+    ) public {
+        ValidationData storage data = validationData[id];
+
+        /**
+         * @dev verify that sender is the same as in `newSignatureCommitment`
+         */
+        require(msg.sender == data.senderAddress, "Error: Sender address does not match original validation data");
+
+        uint256 requiredNumOfSignatures = validatorRegistry.numOfValidators() * THRESHOLD_NOM / THRESHOLD_DENOM;
+
+        /**
+         * @dev verify that required number of signatures, positions, public keys and merkle proofs are
+         * submitted
+         */
+        require(signatures.length == requiredNumOfSignatures, "Error: Number of signatures does not match required");
+        require(validatorPositions.length == requiredNumOfSignatures,
+            "Error: Number of validator positions does not match required");
+        require(validatorPublicKeys.length == requiredNumOfSignatures,
+            "Error: Number of validator public keys does not match required");
+        require(validatorPublicKeyMerkleProofs.length == requiredNumOfSignatures,
+            "Error: Number of validator public keys does not match required");
+
+        /**
+         * @dev Generate an array of numbers
+         */
+        uint256[] memory randomBitfield = Bitfield.randomNBitsFromPrior(getSeed(data), data.validatorClaimsBitfield,
+            requiredNumOfSignatures);
+
+        /**
+         *  @dev For each randomSignature, do:
+         */
+        for (uint256 i = 0; i < requiredNumOfSignatures; i++) {
+            /**
+             * @dev Check if validator in randomBitfield
+             */
+            require(
+                randomBitfield.isSet(validatorPositions[i]), "Error: Validator must be once in bitfield"
+            );
+
+            /**
+             * @dev Remove validator from randomBitfield such that no validator can appear twice in signatures
+             */
+            randomBitfield.clear(validatorPositions[i]);
+
+            /**
+             * @dev Check if merkle proof is valid
+             */
+            require(
+                validatorRegistry.checkValidatorInSet(validatorPublicKeys[i], validatorPositions[i],
+                validatorPublicKeyMerkleProofs[i]), "Error: Validator must be in validator set at correct position"
+            );
+
+            /**
+             * @dev Check if signature is correct
+             */
+            require(ECDSA.recover(payload, signatures[i]) == validatorPublicKeys[i], "Error: Invalid Signature");
+        }
+
+        /**
+         * @todo Release the sender stake as collateral
+         */
+        // TODO
+
+        /**
+         * @follow-up Do we need a try-catch block here?
```

Comment:
lets worry about this in the next PR where we implement processPayload properly

Reply:
👍